src/Java.Interop.BootstrapTasks/Java.Interop.BootstrapTasks/JdkInfo.cs

```diff
@@ -40,41 +40,47 @@ public override bool Execute ()
 			Log.LogMessage (MessageImportance.Low, $"  {nameof (MaximumJdkVersion)}: {MaximumJdkVersion}");
 			Log.LogMessage (MessageImportance.Low, $"  {nameof (PropertyFile)}: {PropertyFile}");
 
-			var maxVersion  = GetMaxJdkVersion ();
+			var maxVersion      = GetVersion (MaximumJdkVersion);
+
+			string jarPath      = null;
+			string javacPath    = null;
+			string jdkJvmPath   = null;
+			string includePath  = null;
+
 			var java_home   = GetJavaHomePathFromEnvironment ();
-			if (java_home != null) {
-				var java_home_v = GetVersionFromPath (java_home);
-				if (maxVersion != null && java_home_v != null && java_home_v > maxVersion) {
-					Log.LogMessage (MessageImportance.Low, $"  Skipping JAVA_HOME default value of `{java_home}` as it exceeds MaximumJdkVersion={MaximumJdkVersion}.");
-					java_home = null;
-				}
-				if (java_home != null && !Directory.Exists (Path.Combine (java_home, "include"))) {
-					Log.LogMessage (MessageImportance.Low, $"  Skipping JAVA_HOME default value of `{java_home}` as it does not contain an `include` subdirectory.");
-					java_home = null;
-				}
+			if (java_home != null &&
```

Comment:
the `java_home != null` check is redundant, since `ValidateJdkPath` repeats it. It might look neater if the null check is completely removed